crates/bevy_render/src/render_phase/draw.rs

```diff
@@ -82,15 +82,15 @@ impl<P: PhaseItem> DrawFunctionsInternal<P> {
 
     /// Adds the [`Draw`] function and associates it to the type `T`
     pub fn add_with<T: 'static, D: Draw<P>>(&mut self, draw_function: D) -> DrawFunctionId {
+        let id = DrawFunctionId(self.draw_functions.len().try_into().unwrap());
         self.draw_functions.push(Box::new(draw_function));
-        let id = DrawFunctionId(self.draw_functions.len() - 1);
```

Comment:
Was the removal of `- 1` intentional?

Reply:
Yes. We moved it before the push. This just tends to make it cleaner without the - 1.